content/blog/gargle-1-2-0/index.Rmd

```diff
@@ -0,0 +1,243 @@
+You might see these nicknames in the output of `gargle::gargle_oauth_sitrep()`.
+
+When gargle encounters a token created with the old built-in client, the token is deleted from the cache.
+So don't be surprised if you see some messages about cleaning out (and relocating) the cache, after you upgrade gargle.
+
+You can expect to do the interactive OAuth dance to obtain new credentials, with the new client, stored in the new token cache, approximately once per wrapper package / API.
+
+**If the rolling of the tidyverse OAuth client is highly disruptive to your workflow, consider this a wake-up call** that you should be using your own OAuth client or, quite possibly, an entirely different method of auth.
+Our credential rolling will have no impact on users who use their own OAuth client or service account tokens.
+
+As always, these articles explain how to take more control of auth:
+
+* <https://gargle.r-lib.org/articles/get-api-credentials.html>
+* <https://gargle.r-lib.org/articles/non-interactive-auth.html>
+
+## Workload identity federation
+
+Workload identity federation is a new (as of April 2021) keyless authentication mechanism offered by Google.
+This allows applications running on a non-Google Cloud platform, such as AWS, to access Google Cloud resources without using a conventional service account token, eliminating the security problem posed by long-lived, powerful service account credential files.
+Basically, instead of storing sensitive information in a file that must be managed with great care, the necessary secrets are obtained on-the-fly and exchanged for short-lived tokens, with very granular control over what actions are allowed.
+There is a cost, of course, which is that this auth method requires substantial configuration on both the GCP and AWS sides.
+
+`credentials_external_account()` is a new function that implements workload identity federation.
+`credentials_external_account()` has been inserted into the default registry of credential-fetchers tried by `token_fetch()`, which makes it automatically available in packages that use gargle for auth, such as bigrquery. `credentials_app_default()` recognizes the JSON configuration for an external account and passes such a call along to `credentials_external_account()`.
+This means that, when placed in one of the locations searched by the Application Default Credential strategy, an external account configuration file will be automatically discovered.
+
+This new feature is still experimental and currently only supports AWS.
+This blog post provides a good high-level introduction to workload identity federation:
+
+* [Keyless API authentication—Better cloud security through workload identity federation, no service account keys necessary](https://cloud.google.com/blog/products/identity-security/enable-keyless-access-to-gcp-with-workload-identity-federation)
+
+## Out of bound auth
```

Comment:
I sort of think this needs a little graphic similar to the first, but haven't motivated myself to make it yet.